File: Data Structures & Algorithms/house-robber/submission-0.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int N = nums.size(); 

        std::vector<int>dp(N);
        dp[0] = nums[0];
        dp[1] = std::max(nums[0], nums[1]);
        for(int i = 2; i < N; i++){
            dp[i] = std::max(nums[i] + dp[i-2], dp[i-1]);
        }

        return dp[N - 1];
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        last_seen = {}
        longest = 0 
        i = 0

        for j, ch in enumerate(s):
            if ch in last_seen and last_seen[ch] >= i:
                i = last_seen[ch] + 1
            last_seen[ch] = j
            longest = max(longest, j - i + 1)

        return longest
                


            
            





            
                



        